### PR TITLE
kodiPackages.sponsorblock: init at 0.5.0

### LINKED ISSUE #285377: Package request: Kodi sponsorblock addon for youtube and invidious plugin
**Project description**
Kodi sponsorblork addon for youtube. [Github home page](https://github.com/siku2/script.service.sponsorblock).

This 3rd party port is also mentioned in [official ajayyy's SponsorBlock github page](https://github.com/ajayyy/SponsorBlock/wiki/3rd-Party-Ports#kodi).

**Description**
This is an unofficial port of the [SponsorBlock](https://sponsor.ajay.app/) browser extension. It works as an extension to the [YouTube Plugin](https://github.com/anxdpanic/plugin.video.youtube).

Once installed, the add-on will automatically skip sponsor segments in all YouTube and Invidious videos you watch.

**Metadata**

* homepage URL: https://github.com/siku2/script.service.sponsorblock
* source URL: https://github.com/siku2/script.service.sponsorblock/archive/refs/tags/v0.5.0.tar.gz
* license: mit
* platforms: unix, linux, darwin, ...

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by amalgame21**
## Description of changes
Closes #285377

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/kodi/addons/sponsorblock/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    requests
+  ];
+
+  passthru = {
+    pythonPath = "resources/lib";
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/siku2/script.service.sponsorblock";
+    description = "An unofficial port of the SponsorBlock browser extension. It works as an extension to the YouTube Plugin. Once installed, the add-on will automatically skip sponsor segments in all YouTube and Invidious videos you watch.";
```
Comment (aanderse) [inline pkgs/applications/video/kodi/addons/sponsorblock/default.nix:25]: `description` should be very short, please see other packages for examples
you want `longDescription`

**Review by aanderse — CHANGES_REQUESTED** [review]
> * Tested basic functionality of all binary files (usually in `./result/bin/`)

can you please test to make sure this runs?

File: pkgs/applications/video/kodi/addons/sponsorblock/default.nix (hunk 1 above)
Comment (amalgame21) [reply pkgs/applications/video/kodi/addons/sponsorblock/default.nix]: Thanks you for your advice. I have shortened the description.

**Comment by amalgame21** [comment]
> > * Tested basic functionality of all binary files (usually in `./result/bin/`)
> 
> can you please test to make sure this runs?

I am quite sure it runs.
I use my github branch as `flake.nix` input and use it in my `configuration.nix`. After running `nixos-rebuild switch` it build the kodi and kodi-ffmpeg from source, and also this addon is installed correctly.
I am quite new to nixos. May I ask how can I test it in a proper way as a developer rather than directly put it in `configuration.nix`? Thanks.

**Comment by aanderse** [comment]
thanks for your contribution!

**Review by aanderse — APPROVED** [review]

**Pull request merged**